src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private int[] array;
+    private int n;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+        this.n = array.length;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length == 1) {
+            return;
+        }
+        var left = Arrays.copyOfRange(array, 0 , n/2);
```

Comment:
It's more efficient to use n / 2 directly instead of array.length / 2 since you've already stored the length of the array in the variable 'n'.